src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -232,10 +234,24 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 			if (this.cellModel.currentMode !== CellEditModes.WYSIWYG) {
 				needsTransform = false;
 			} else {
+				let result = undefined;
+				const isFile = URI.parse(linkCalloutResult?.insertUnescapedLinkUrl).scheme.includes('file');
+				if (!path.isAbsolute(linkCalloutResult?.insertUnescapedLinkUrl) && isFile) {
+					// Counts the number of ..\ intermediate directories when user enters a relative path
+					let midDirectories = (linkCalloutResult?.insertUnescapedLinkUrl.match(/..\\/g) || []).length;
+					let dirName = path.dirname(this.cellModel?.notebookModel?.notebookUri.fsPath);
+					const relativePath = linkCalloutResult?.insertUnescapedLinkUrl.replace(/\.\\/g, '/');
```

Comment:
This doesn't seem like it'd handle multiple .. well...for example if I have `..\..\..\notebook.ipynb` then wouldn't relative path end up being `./././notebook.ipynb`?